### DISCUSSION TITLE
Is it possible to change text in a specific location in a page?

### DISCUSSION THREAD

**Discussion by ali-asly101** [pymupdf/PyMuPDF#1613] [Looking for help]
I am developing a PDF editor app for school exams and one of the features I want to add is being able to change the question numbers.

These question numbers are usually on the very left on the page, and if I do a for loop looking for like lets say the number 2 and change it to 1, it can also change the questions' content itself ( Lets say a question says " John has 2 oranges ", it will also change these 2 oranges to 1, ruining the question ). So is it possible to do a for loop looking for specific texts in a specific rect / position of the page?

Thanks!

**Comment by JorjMcKie** [comment]
Yes, this works by using redaction annotations.
1. determine the rectangle containing the text to be replaced
2. define `annot = page.add_redact_annot(rect, text="newtext", ...)`
3. repeat steps 1 and 2 as many times as needed.
4. execute `page.apply_redactions()`

For short text, and text not longer than the original, this should work ok. There may be unpleasant misplacements if new text is longer, or using a different font / fontsize, etc.

There are ways to improve the result. Drop a note when you encounter things like this.

**Reply by ali-asly101** [reply]
thanks I will try it
